filebeat/prospector/log/harvester.go

```diff
@@ -111,3 +125,381 @@ func (h *Harvester) open() error {
 func (h *Harvester) ID() uuid.UUID {
 	return h.id
 }
+
+// Setup opens the file handler and creates the reader for the harvester
+func (h *Harvester) Setup() error {
+	err := h.open()
+	if err != nil {
+		return fmt.Errorf("Harvester setup failed. Unexpected file opening error: %s", err)
+	}
+
+	h.reader, err = h.newLogFileReader()
+	if err != nil {
+		if h.source != nil {
+			h.source.Close()
+		}
+		return fmt.Errorf("Harvester setup failed. Unexpected encoding line reader error: %s", err)
+	}
+
+	return nil
+
+}
+
+// Harvest reads files line by line and sends events to the defined output
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported method Harvester.Run should be of the form "Run ..."